core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksInodeStore.java

```diff
@@ -0,0 +1,298 @@
+  private static final String EDGES_COLUMN = "edges";
+
+  private final String mBaseDir;
+
+  private String mDbPath;
+  private RocksDB mDb;
+  private ColumnFamilyHandle mDefaultColumn;
+  private ColumnFamilyHandle mInodesColumn;
+  private ColumnFamilyHandle mEdgesColumn;
+  private ColumnFamilyHandle mLastModifiedColumn;
+
+  /**
+   * Creates and initializes a rocks block store.
+   */
+  public RocksInodeStore() throws RocksDBException {
+    mBaseDir = Configuration.get(PropertyKey.MASTER_METASTORE_DIR);
+    RocksDB.loadLibrary();
+    initDb();
+  }
+
+  @Override
+  public void remove(InodeView inode) {
+    try {
+      mDb.delete(mInodesColumn, Longs.toByteArray(inode.getId()));
+      mDb.deleteRange(mEdgesColumn, Longs.toByteArray(inode.getId()),
+          Longs.toByteArray(inode.getId() + 1));
+      mDb.delete(mLastModifiedColumn, Longs.toByteArray(inode.getId()));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void setModificationTimeMs(long id, long opTimeMs) {
+    try {
+      mDb.put(mLastModifiedColumn, Longs.toByteArray(id), Longs.toByteArray(opTimeMs));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void writeInode(Inode<?> inode) {
+    try {
+      mDb.put(mInodesColumn, Longs.toByteArray(inode.getId()), inode.toProto().toByteArray());
+      mDb.put(mLastModifiedColumn, Longs.toByteArray(inode.getId()),
+          Longs.toByteArray(inode.getLastModificationTimeMs()));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void clear() {
+    try {
+      initDb();
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void addChild(long parentId, InodeView inode) {
+    try {
+      mDb.put(mEdgesColumn, RocksUtils.toByteArray(parentId, inode.getName()),
+          Longs.toByteArray(inode.getId()));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeChild(long parentId, String name) {
+    try {
+      mDb.delete(mEdgesColumn, RocksUtils.toByteArray(parentId, name));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public int estimateSize() {
+    try {
+      return Integer.parseInt(mDb.getProperty(mInodesColumn, "rocksdb.estimate-num-keys"));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Optional<InodeView> get(long id) {
+    byte[] inode;
+    try {
+      inode = mDb.get(mInodesColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (inode == null) {
+      return Optional.empty();
+    }
+    try {
+      return Optional.of(Inode.fromProto(InodeMeta.Inode.parseFrom(inode), getLastModifiedMs(id)));
```

Comment:
Updating parent modification times during create, delete, and rename operations requires reading the parent inode, updating the last modified time, then overwriting the full inode. This is kind of sketchy since we're overwriting the inode, but only holding the read lock. It also makes rename more expensive since now we need to read/modify/update the full parent inode.

However, I can't think of anything that will break due to updating the parent last modified while holding the read lock, as long as nothing else tries to modify another field while holding the read lock. And reducing from 2 calls to 1 in the most common function call is valuable. I've moved the last modified time into the inode protobuf so that we only need 1 call